sklearn/cluster/tests/test_dbscan.py

```diff
@@ -78,6 +77,19 @@ def test_dbscan_sparse():
     assert_array_equal(labels_dense, labels_sparse)
 
 
+def test_dbscan_no_score_samples():
+    rng = np.random.RandomState(0)
+    X = rng.rand(40, 10)
+    X[X < .8] = 0
+
+    db = DBSCAN().fit(X)
+    assert_array_equal(db.components_, np.empty((0, X.shape[1])))
```

Comment:
I assume that all samples are clustered as outliers in that case? It would be great to add an assertion to check that here, first by asserting that `core_sample_indices_` has shape `(0,)` and second that `labels_` is filed with `-1`s.